server/src/main/java/com/google/sps/utils/AdRowProcessor.java

```diff
@@ -0,0 +1,129 @@
+        .setDatabaseUrl(DATABASE_URL)
+        .build();
+    if(FirebaseApp.getApps().isEmpty()) {
+      FirebaseApp.initializeApp(options);
+    }
+  }
+
+  public void addAdToDatabase(Ad ad, int rowIndex, String collection) throws Exception {
+    Firestore db = FirestoreClient.getFirestore();    
+    ApiFuture<WriteResult> result = db.collection(collection).document(ad.id).set(ad);
+    System.out.println("Update time : " + result.get().getUpdateTime());
+  }
+
+  public Ad createAd() {
+    Ad ad = Ad.newBuilder()
+        .id(row[0])
+        .advertiser(row[1])
+        .startDate(row[2]) 
+        .endDate(row[3])
+        .impressionsMin(getImpressionsMin(row[4]))
+        .impressionsMax(getImpressionsMax(row[4]))
+        .isTargetingAge(getAgeTargets(row[5]))
+        .genderTargets(convertStringToList(row[6]))
+        .geoTargets(convertStringToList(row[7]))
+        .spendMin(convertStringToLong(row[8]))
+        .spendMax(convertStringToLong(row[9]))
+        .headline(row[10].trim())
+        .link(row[11].substring(3)) // trim "Ad" from "Ad {URL}"
+        .content(row[12].trim())
+        .headlineSentiment(row[13].trim())
+        .headlineTerms(row[14].trim())
+        .contentSentiment(row[15].trim())
+        .contentTerms(row[16].trim())
+        .build();
+    return ad; 
+  }
+
+  /* Min and max impression fields contain letters k and M as well as ≤. The 
+  formatImpressionField helper method makes the fields easier to parse later.
+  */ 
+  public String[] formatImpressionField(String str) {
+    String impressionsString = str.replace("k", "000").replace("M", "000000").replace(" ","").replace("≤", "");
+    String[] impressionsArray = impressionsString.split("-");
+    return impressionsArray; 
+  }
+
+  public long getImpressionsMin(String impressionsField) throws IllegalArgumentException{
+    String[] impressionsArray = formatImpressionField(impressionsField); 
+    // If csv field uses "<=", arr has length 1 and min number of impressions is 0.
+    if (impressionsArray.length > 1) {
+      try {
+        return Long.parseLong(impressionsArray[0]);
+      } catch (IllegalArgumentException e) {
+        return -1; 
+      }
+    }
+    return 0;
+  }
+
+  public long getImpressionsMax(String impressionsField) throws IllegalArgumentException {
+    String[] impressionsArray = formatImpressionField(impressionsField); 
+    try {
+      return Long.parseLong(impressionsArray[impressionsArray.length - 1]);   
+    } catch (IllegalArgumentException e) {
+      return -1; 
+    }
+  }
+
+  public boolean getAgeTargets(String str) {
+    if (str.trim().equals("Not targeted")) {
+      return false; 
+    }
+    return true; 
+  }
+
+  public List<String> convertStringToList(String str) {
+    List<String> trimmedList = Arrays.stream(str.split(","))
+        .map(String::trim)
+        .collect(Collectors.toList());
+    return trimmedList;  
+  }
+
+  public long convertStringToLong(String str) {
```

Comment:
Is this expected in the data?

Reply:
I receive strings like "100000" that I need to save to the db as longs.